test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/MaxRequestBodySizeTests.cs

```diff
@@ -0,0 +1,384 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class MaxRequestBodySizeTests
+    {
+        [Fact]
+        public async Task RejectsRequestWithContentLengthHeaderExceedingGlobalLimit()
+        {
+            // 4 GiB
+            var globalMaxRequestBodySize = 0x100000000;
+            BadHttpRequestException requestRejectedEx = null;
+
+            using (var server = new TestServer(async context =>
+            {
+                var buffer = new byte[1];
+                requestRejectedEx = await Assert.ThrowsAsync<BadHttpRequestException>(
+                    async () => await context.Request.Body.ReadAsync(buffer, 0, 1));
+                throw requestRejectedEx;
+            },
+            new TestServiceContext { ServerOptions = { Limits = { MaxRequestBodySize = globalMaxRequestBodySize } } }))
+            {
+                using (var connection = server.CreateConnection())
+                {
+                    await connection.Send(
+                        "POST / HTTP/1.1",
+                        "Host:",
+                        "Content-Length: " + (globalMaxRequestBodySize + 1),
+                        "",
+                        "");
+                    await connection.ReceiveForcedEnd(
+                        "HTTP/1.1 413 Payload Too Large",
```

Comment:
The BadHttpRequestException is catchable by the app, correct? You only get this 413 if they do not catch it and it bubbles back out from the app pipeline?